db.go

```diff
@@ -492,31 +423,50 @@ func (db *DB) reload(deleteable ...string) (err error) {
 	if err != nil {
 		return errors.Wrap(err, "find blocks")
 	}
+	// We delete old blocks that have been superseded by new ones by gathering all parents
```

Comment:
The comment should mention how this can happen, so that future readers know this is designed to fix the relevant race condition. A matching comment should go in the delete/rename compaction code pointing to here.